src/main/java/core/basesyntax/serviceimpl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String REGEX_TO_SPLIT = ",";
+    private static final String CONTENTS = "fruit, quantity";
+
+    @Override
+    public String createReport() {
+        StringBuilder stringBuilder = new StringBuilder(CONTENTS + System.lineSeparator());
+        for (Map.Entry<String, Integer> entry : FruitStorage.getFruits().entrySet()) {
+            stringBuilder.append(entry.getKey())
+                    .append(REGEX_TO_SPLIT)
+                    .append(entry.getValue())
+                    .append(System.lineSeparator());

```

Comment:
According to the encapsulation principle, direct access to `FruitStorage.getFruits()` should be avoided. It would be better to have a service method that provides the necessary data without exposing the storage implementation.